contracts/peripheral/IncurDebt.sol

```diff
@@ -383,23 +367,27 @@ contract IncurDebt is OlympusAccessControlledV2, IIncurDebt {
         ITreasury(treasury).repayDebtWithOHM(ohmToRepay);
     }
 
-    /// @notice withdraws sOHM  to _to address
+    /// @notice withdraws gOHM
     /// - msg.sender must be a borrower
     /// - _amount (in OHM) must be less than or equal to depositedOhm - debt
-    /// @param _amount amount of sOHM to withdraw
-    /// @param _to address to send _amount
-    function withdraw(uint256 _amount, address _to) external override isBorrower(msg.sender) {
+    /// @param _amount amount of gOHM to withdraw
+    function withdraw(uint256 _amount) external override isBorrower(msg.sender) {
         if (_amount == 0) revert IncurDebt_InvaildNumber(_amount);
         Borrower storage borrower = borrowers[msg.sender];
 
-        updateCollateralInSOHM();
         if (_amount > getAvailableToBorrow()) revert IncurDebt_AmountAboveBorrowerBalance(_amount);
 
-        borrower.collateralInSOHM -= uint128(_amount);
-        borrower.borrowerGonBalance -= IsOHM(sOHM).gonsForBalance(_amount);
-        IERC20(sOHM).transfer(msg.sender, _amount);
+        if (_amount > borrower.collateralInGOHM - borrower.unwrappedGOHM) { // Does uint256 > uint128 cause problem?
+            uint256 amountGOHMToWrap = borrower.unwrappedGOHM + _amount - borrower.collateralInGOHM;
+            borrower.unwrappedGOHM -= uint128(amountGOHMToWrap);
```

Comment:
This will always fail say;
`borrower.collateralInGOHM` = 10 gOHM
`borrower.unwrappedGOHM` = 5 gOHM
`amount` = 11 gOHM
The calculation from line `381` will equal to 6
Resulting to 5 -6

Reply:
Not sure how to rename variables to make this clearer but
collateralInGOHM is meant to be their total collateral value in gOHM terms
unwrappedGOHM is meant to represent how much of their collateralInGOHM is unwrapped in sOHM. 

so unwrappedGOHM <= collateralInGOHM

so max amount is = collateralInGOHM which is >= unwrappedGOHM